### PR TITLE
Disable servicespec selection on service creation

### PR DESCRIPTION
**Pull request by drshrey**
This PR does a few things:
- Removes ServiceSpecID param from Service creation
- Generates a service spec that derives properties from default service spec, for each service
- makes service spec mutations atomic

### CODE REVIEW

File: plugins/codeamp/db/service.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func (r *ServiceResolver) Name() string {
 func (r *ServiceResolver) ServiceSpec() *ServiceSpecResolver {
 	var serviceSpec model.ServiceSpec
 
-	r.DB.Model(r.Service).Related(&serviceSpec)
+	r.DB.Where("service_id = ?", r.Service.Model.ID).First(&serviceSpec)
```
Comment (aballman) [inline plugins/codeamp/db/service.go]: Can we add some error handling here? At the very least we should log the error if there is one.

File: plugins/codeamp/graphql/project_mutation.go [hunk 2]
```diff
@@ -116,6 +114,8 @@ func (r *ProjectResolverMutation) CreateProject(ctx context.Context, args *struc
 		return nil, fmt.Errorf("No envs found")
 	}
 
+	r.DB.Create(&project)
```
Comment (aballman) [inline plugins/codeamp/graphql/project_mutation.go]: I don't mind if we do this in another PR, but we should probably add some error checking here as well on the create operation itself.

File: plugins/codeamp/graphql/project_test.go [hunk 3]
```diff
@@ -276,6 +279,7 @@ func (suite *ProjectTestSuite) TestCreateProjectFailure() {
 	}
 
 	// Project
+	suite.helper.CreateProjectWithInput(suite.T(), &projectInput)
```
Comment (aballman) [inline plugins/codeamp/graphql/project_test.go]: Is this intentional to create two here?

File: plugins/codeamp/graphql/service_mutation.go [hunk 4]
```diff
@@ -90,7 +90,20 @@ func (r *ServiceResolverMutation) CreateService(args *struct{ Service *model.Ser
 		PreStopHook:        preStopHook,
 	}
 
-	r.DB.Create(&service)
+	r.DB.Create(&service)
```
Comment (aballman) [inline plugins/codeamp/graphql/service_mutation.go]: Check for error on create

File: plugins/codeamp/graphql/project_mutation.go (hunk 2 above)
Comment (drshrey) [reply plugins/codeamp/graphql/project_mutation.go]: yup, created an issue here: https://github.com/codeamp/circuit/issues/380
